src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/TimedFlipFlopDetector.java

```diff
@@ -0,0 +1,114 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.collections.TimeExpiringSet;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Impact;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A {@link FlipFlopDetector} whose recorded actions expire after a given period of time.
+ *
+ * <p>This class defines a flip flop as an {@link Impact#DECREASES_PRESSURE}s followed by an
+ * {@link Impact#INCREASES_PRESSURE}s to be a flip flops.
+ *
+ * <p>This class stores a {@link TimeExpiringSet} of {@link ImpactVector}s per {@link NodeKey}
+ * that are used to determine these flip flops.
+ */
+public class TimedFlipFlopDetector implements FlipFlopDetector {
+    private Map<NodeKey, TimeExpiringSet<ImpactVector>> flipFlopMap;
+    private long expiryDuration;
+    private TimeUnit expiryUnit;
+
+    public TimedFlipFlopDetector(long duration, TimeUnit unit) {
+        flipFlopMap = new HashMap<>();
+        this.expiryDuration = duration;
+        this.expiryUnit = unit;
+    }
+
+    /**
+     * Tests if (a,b) is a flip flopping sequence of impacts.
+     *
+     * <p>Only an increase following a decrease is considered a flip flop
+     *
+     * @param a The first impact that would be applied
+     * @param b The subsequent impact that would be applied
+     * @return Whether or not (a,b) is a flip flopping sequence of impacts
+     */
+    protected boolean isFlipFlopImpact(Impact a, Impact b) {
+        return a.equals(Impact.DECREASES_PRESSURE) && b.equals(Impact.INCREASES_PRESSURE);
+    }
+
+    /**
+     * Returns true if the impact for a given Dimension in v is a flip flop Impact when compared to
+     * the impact for a given dimension in u
+     *
+     * <p>e.g. for u = (HEAP: INCREASE, CPU: DECREASE), v = (HEAP: DECREASE, CPU: INCREASE)
+     * (u,v) is a flip flop vector because a CPU: DECREASE followed by a CPU: INCREASE is a flip
+     * flop impact
+     *
+     * @param u The first impact vector that would be applied
```

Comment:
For the system, the first impact vector has already been applied, right? It's the second one that we are now evaluating for *flip-flopping*. I understand that writing it this way keeps this method independent of the system, but it's also confusing. Pls consider renaming/modifying the doc strings.